src/compiler/transformers/rewrite_aliased_paths.ts

```diff
@@ -0,0 +1,152 @@
+import { normalizePath } from '@utils';
+import { dirname, relative } from 'path';
+import ts from 'typescript';
+
+import { retrieveTsModifiers } from './transform-utils';
+
+/**
+ * Transform modules aliased with `paths` in `tsconfig.json` to relative
+ * imported in `.d.ts` files.
+ *
+ * @returns a TypeScript transformer factory
+ */
+export function rewriteAliasedDTSPaths(): ts.TransformerFactory<ts.Bundle | ts.SourceFile> {
+  return (transformCtx: ts.TransformationContext) => {
+    const compilerHost = ts.createCompilerHost(transformCtx.getCompilerOptions());
+
+    return (tsBundleOrSourceFile) => {
+      const fileName = ts.isBundle(tsBundleOrSourceFile)
+        ? tsBundleOrSourceFile.getSourceFile().fileName
+        : tsBundleOrSourceFile.fileName;
+
+      return ts.visitEachChild(tsBundleOrSourceFile, visit(compilerHost, transformCtx, fileName), transformCtx);
+    };
+  };
+}
+
+/**
+ * Transform modules aliased with `paths` in `tsconfig.json` to relative
+ * imported in source files.
+ *
+ * @returns a TypeScript transformer factory
+ */
+export const rewriteAliasedSourceFilePaths = (): ts.TransformerFactory<ts.SourceFile> => {
```

Comment:
Same idea as above: `rewriteAliasedSourceFileImportPaths`

Also, just feels strange one of these is a `function` and one a `const`... makes no functional difference, just something I noticed

Reply:
sure, consistency is fine by me!